linode/manager
# test: [M3-7498] - Add Cypress tests for Parent/Child billing page UX enhancements

## Description 📝
Adds Cypress tests to cover the UX enhancements made to the account Billing page for restricted and child users. Additionally updates Vitest and enables a Parent/Child unit test that was disabled due to an issue in older Vitest versions.

## Changes  🔄
- Upgrades Vitest from `v1.0.4` to `v1.2.0`
- Enables commented out unit test in `ContactInformation.test.tsx`
- Adds Cypress tests to cover UX enhancements made to billing page for Parent/Child Account project
  - Confirms action buttons are disabled for restricted and child users
  - Confirms action buttons are enabled for parent and regular users

## How to test 🧪
We should be able to rely on automated tests for this PR. Refer to the instructions below if you'd like to run the tests locally, however:

### Prerequisites
Build and serve a Cloud Manager production build
```bash
yarn && yarn build && yarn start:manager:ci
```

### Verification steps 
1. Verify unit test change:
   
   ```bash
   yarn test ContactInformation.test
   ```
2. Verify Cypress test additions:
   
   ```bash
   yarn cy:run -s "cypress/e2e/core/billing/restricted-user-billing.spec.ts"
   ```

## As an Author I have considered 🤔

*Check all that apply*
